core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -118,20 +126,22 @@ trait GraphQL[-R] { self =>
                                                                                 config.skipValidation,
                                                                                 config.validations
                                                                               )
+                                                              _            <- ZIO.unless(config.allowMutationsOverGetRequests)(checkHttpMethod(executionReq))
```

Comment:
Can you check if it changes anything to benchmarks?

Reply:
I couldn't notice any difference in performance, but just to be absolutely sure, I ended up refactoring it a bit so that we only get the data from the `FiberRef` when the operation is a mutation. For queries, the method should be a no-op essentially